state.go

```diff
@@ -1023,7 +1023,14 @@ func (s *State) OnInterface(se *Session, i interface{}) (err error) {
 		}
 	case *GuildEmojisUpdate:
 		if s.TrackEmojis {
-			err = s.EmojisAdd(t.GuildID, t.Emojis)
+			var guild *Guild
+			guild, err = s.Guild(t.GuildID)
+			if err != nil {
+				return err
+			}
+			s.Lock()
+			defer s.Unlock()
+			guild.Emojis = t.Emojis
```

Comment:
Additionally, after looking at more functions in the `state`, I think it would have made more sense to actually separate it out (that is why I deleted the original comment), like in `ThreadMemberUpdate`. But since change would not be significant, hope you don't mind if I make it myself.